ecs_client_test.go

```diff
@@ -39,49 +60,205 @@ func TestECSClientRegisterAndDeregisterTaskDefinition(t *testing.T) {
-		require.NotZero(t, out)
+				Cpu:    aws.String("128"),
+				Memory: aws.String("4"),
+				Family: aws.String(os.Getenv("AWS_ECS_TASK_DEFINITION_PREFIX") + "foo"),
+			})
+			require.NoError(t, err)
+			require.NotZero(t, out)
+
+			defer func() {
+				if out != nil && out.TaskDefinition != nil && out.TaskDefinition.TaskDefinitionArn != nil {
+					out, err := c.DeregisterTaskDefinition(ctx, &ecs.DeregisterTaskDefinitionInput{
+						TaskDefinition: out.TaskDefinition.TaskDefinitionArn,
+					})
+					require.NoError(t, err)
+					require.NotZero(t, out)
+				}
+			}()
+
+			require.NotZero(t, out.TaskDefinition)
+			require.NotZero(t, out.TaskDefinition.Status)
+			assert.Equal(t, "ACTIVE", *out.TaskDefinition.Status)
+			require.NotZero(t, out.TaskDefinition.RegisteredAt)
+			assert.NotZero(t, *out.TaskDefinition.RegisteredAt)
+		},
+		"RunTaskFailsWithInvalidInput": func(ctx context.Context, t *testing.T, c *BasicECSClient) {
+			out, err := c.RunTask(ctx, &ecs.RunTaskInput{})
+			assert.Error(t, err)
+			assert.Zero(t, out)
+		},
+		"DescribeTasksFailsWithInvalidInput": func(ctx context.Context, t *testing.T, c *BasicECSClient) {
+			out, err := c.DescribeTasks(ctx, &ecs.DescribeTasksInput{})
+			assert.Error(t, err)
+			assert.Zero(t, out)
+		},
+		"StopTaskFailsWithInvalidInput": func(ctx context.Context, t *testing.T, c *BasicECSClient) {
+			out, err := c.StopTask(ctx, &ecs.StopTaskInput{})
+			assert.Error(t, err)
+			assert.Zero(t, out)
+		},
+		"RunTaskFailsWithValidButNonexistentInput": func(ctx context.Context, t *testing.T, c *BasicECSClient) {
+			out, err := c.RunTask(ctx, &ecs.RunTaskInput{
+				TaskDefinition: aws.String(utility.RandomString()),
+			})
+			assert.Error(t, err)
+			assert.Zero(t, out)
+		},
+		"DescribeTasksFailsWithNoClusterAndValidButNonexistentInput": func(ctx context.Context, t *testing.T, c *BasicECSClient) {
+			out, err := c.DescribeTasks(ctx, &ecs.DescribeTasksInput{
+				Tasks: []*string{aws.String(utility.RandomString())},
+			})
+			assert.Error(t, err)
+			assert.Zero(t, out)
+		},
+		"DescribeTasksReturnsFailureWithClusterAndValidButNonexistentInput": func(ctx context.Context, t *testing.T, c *BasicECSClient) {
+			out, err := c.DescribeTasks(ctx, &ecs.DescribeTasksInput{
+				Cluster: aws.String(os.Getenv("AWS_ECS_CLUSTER")),
+				Tasks:   []*string{aws.String("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa")},
+			})
+
+			assert.NoError(t, err)
+			assert.NotZero(t, out)
+			assert.NotZero(t, out.Failures)
+			assert.Empty(t, out.Tasks)
+
+		},
+		"StopTaskFailsWithValidButNonexistentInput": func(ctx context.Context, t *testing.T, c *BasicECSClient) {
+			out, err := c.StopTask(ctx, &ecs.StopTaskInput{
+				Task: aws.String(utility.RandomString()),
+			})
+			assert.Error(t, err)
+			assert.Zero(t, out)
+		},
+		"RegisterSucceedsWithDuplicateTaskDefinition": func(ctx context.Context, t *testing.T, c *BasicECSClient) {
+			outDuplicate, err := c.RegisterTaskDefinition(ctx, &ecs.RegisterTaskDefinitionInput{
```

Comment:
Rather than copy-paste this task definition in two different places (once here, once during the setup code), can you just assign it to a variable and pass it to `RegisterTaskDefinition`?